src/document.jl

```diff
@@ -100,11 +100,8 @@ function get_offset2(doc::Document, line::Integer, character::Integer)
     line_offsets = get_line_offsets2!(doc)
     text = get_text(doc)
 
-    if line >= length(line_offsets)
-        error("Invalid arguments.")
-        return nextind(text, lastindex(text))
-    elseif line < 0
-        error("Invalid arguments.")
+    if line >= length(line_offsets) || line < 0
+        throw(LSOffsetError("get_offset crashed. More diagnostics:\nline=$line\ncharacter=$character\nline_offsets='$(line_offsets)'\ntext='$(obscure_text(get_text(doc)))'"))
```

Comment:
I wonder why I had that `return` statement in there... I think the code I copied from the node implementation might actually have returned that and not an error? Maybe we leave that return statement in there for now, just to remind us that I might have had a reason for it, even if it can't be reached right now.

Reply:
okay, I will add a small comment aside as well.